packages/next/src/executors/build/build.impl.ts

```diff
@@ -22,18 +26,24 @@ export default async function buildExecutor(
   options: NextBuildBuilderOptions,
   context: ExecutorContext
 ) {
+  let dependencies: DependentBuildableProjectNode[] = [];
   process.env.NODE_ENV = process.env.NODE_ENV || 'production';
 
   const root = resolve(context.root, options.root);
 
-  const projGraph = createProjectGraph();
-  const { dependencies } = calculateProjectDependencies(
-    projGraph,
-    context.root,
-    context.projectName,
-    context.targetName,
-    context.configurationName
-  );
+  if (!options.buildLibsFromSource && context.targetName) {
+    const projGraph = createProjectGraph();
+    const result = calculateProjectDependencies(
+      projGraph,
+      context.root,
+      context.projectName,
+      context.targetName,
+      context.configurationName
+    );
+    dependencies = result.dependencies;
+
+    assertDependentProjectsHaveBeenBuilt(dependencies, context);
+  }
 
   const config = await prepareConfig(
```

Comment:
This function eventually calls `createTmpTsconfig` which should also not be done if the user is building from source.

Reply:
ah, you're right.. :/ but we need to call `prepareConfig` either way.. so i think i'll change it so that we _conditionally_ call `createTmpTsconfig` - only when `dependencies` are not empty

Reply:
added a conditional to `prepareConfig` internally